src/nccl_ofi_rdma.c

```diff
@@ -4262,7 +4293,7 @@ static int post_bounce_buffer(nccl_net_ofi_rdma_req_t *req,
 	rdma_req_bounce_data_t *bounce_data = get_bounce_data(req);
 	nccl_net_ofi_rdma_bounce_fl_item_t *bounce_fl_item = bounce_data->bounce_fl_item;
 	freelist_regmr_fn_handle_t *fl_mr_handle = bounce_fl_item->fl_reginfo.mr_handle;
-	void *desc = fi_mr_desc(fl_mr_handle->mr_handle->mr[bounce_data->bounce_rail_id]);
+	void *desc = fi_mr_desc(fl_mr_handle->mr_handle->mr[bounce_data->rail->rail_id]);
```

Comment:
This is another place where you should selectively use `control_mr` instead of rail 0 `mr`. Though, as you mention, it doesn't matter for EFA provider.

Reply:
I'm not sure I follow.  Bounce buffers have a back pointer to the rail that they cam from, and during `post_bounce_buffs()`, the control messages are created with the pointer to the control rail, so the mr will be from the right endpoint.

Reply:
In this PR, you define `mr_handle` such that it has a separate `control_mr` and an `mr` array, one for each data rail mr. So, in the line above, `mr[bounce_data->rail->rail_id]` will be data rail 0 mr, rather than the control mr.

Reply:
Yeah, fair.  Added some comments and removed support for FI_MR_ENDPOINT in the RDMA transport.